activerecord/lib/active_record/secure_token.rb

```diff
@@ -0,0 +1,38 @@
+module ActiveRecord
+  module SecureToken
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      # Example using has_secure_token
+      #
+      #   # Schema: User(auth_token:string, invitation_token:string)
+      #   class User < ActiveRecord::Base
+      #     has_secure_token
+      #   end
+      #
+      #   user = User.new
+      #   user.save
+      #   user.token # => "44539a6a59835a4ee9d7b112"
+      #   user.regenerate_token # => true
+      def has_secure_token(attribute = :token)
+        # Load securerandom only when has_secure_key is used.
+        require 'securerandom'
+        define_method("regenerate_#{attribute}") { update! attribute => self.class.generate_unique_secure_token(attribute) }
+        before_create { self.send("#{attribute}=", self.class.generate_unique_secure_token(attribute)) }
+      end
+
+      def generate_unique_secure_token(attribute)
+        10.times do |i|
+          SecureRandom.hex(12).tap do |token|
+            if exists?(attribute => token)
```

Comment:
@chancancode yes, I am. I think that the path here is:

1. Add Base62 monkey patch to ActiveSupport
2. Switch to Base62 for generates secure tokens and remove the retry
3. Add the secure tokens migration helper and generator
4. Move SecureToken to ActiveModel, in favor of "plug and play"